corehq/apps/campaign/views.py

```diff
@@ -63,12 +63,20 @@ class DashboardView(BaseProjectReportSectionView, DashboardMapFilterMixin):
     def page_url(self):
         return reverse(self.urlname, args=[self.domain])
 
+    @property
+    def dashboard(self):
+        """
+        Returns the campaign dashboard for the domain. Creates an empty
+        one if it doesn't exist yet.
+        """
+        dashboard, __ = Dashboard.objects.get_or_create(domain=self.domain)
+        return dashboard
+
     def get_context_data(self, **kwargs):
         context = super().get_context_data(**kwargs)
-        dashboard = Dashboard.objects.get(domain=self.domain)
```

Comment:
Can you share why this is better than just handling the exception if the dashboard does not exist? 
I followed that approach here, https://github.com/dimagi/commcare-hq/pull/36065/commits/1081bb038b6f5b0b098fb449b563caa7dfc3ec19

If someone is visiting the campaign dashboard they definitely will go ahead and add widgets?

Reply:
I didn't see any code on Staging that allows someone to create a dashboard using the UI. (Did I miss it?) So I figured this would let them test without needing to use a Python shell first.

Reply:
Ya, it makes sense one can't create a dashboard object from UI.

I was thinking that we could create it along with the first widget itself. I am okay with this as well.